core/frontend/src/render/webgl/Texture.ts

```diff
@@ -555,13 +606,16 @@ export class ExternalTextureLoader { /* currently exported for tests only */
 
     try {
       if (!req.imodel.isClosed) {
-        const maxTextureSize = System.instance.capabilities.maxTextureSize;
-        const texBytes = await req.imodel.getTextureImage({ name: req.name, maxTextureSize });
-        if (undefined !== texBytes) {
-          const imageSource = new ImageSource(texBytes, req.format);
-          const image = await imageElementFromImageSource(imageSource);
+        const maxTextureSize = System.instance.capabilities.maxTexSizeAllow;
+        const texData = await req.imodel.queryTextureImage({ name: req.name, maxTextureSize });
+        if (undefined !== texData) {
+          const cnvReq = { req, texData };
+          this._convertRequests.push(cnvReq);
+          do {

```

Comment:
Can you explain to me how this loop works if `_convertPending` is true? It looks like it exits immediately without trying to convert this texture. So when does this texture get converted?

Reply:
That is what happens, in order to prevent overlapping calls to _convertTexture, which can occur from overlapping calls to _activateRequest.  It is put on the list, so if it doesn't get converted here it will get converted by the loop that is converting the current one.

Reply:
I see now. Please add a comment to that effect for dummies like me.

Reply:
Done